src/compile.ts

```diff
@@ -556,10 +558,17 @@ export function compile(
         ) {
           return toExpr(node.declarationList.declarations[0], scope);
         } else if (ts.isVariableDeclaration(node)) {
-          return newExpr("VariableStmt", [
-            ts.factory.createStringLiteral(node.name.getText()),
-            ...(node.initializer ? [toExpr(node.initializer, scope)] : []),
-          ]);
+          if (ts.isIdentifier(node.name)) {
+            return newExpr("VariableStmt", [
+              ts.factory.createStringLiteral(node.name.getText()),
+              ...(node.initializer ? [toExpr(node.initializer, scope)] : []),
+            ]);
+          } else {
+            return newExpr("VariableStmt", [
+              toExpr(node.name, scope),
+              toExpr(node.initializer, scope),
+            ]);
+          }
```

Comment:
Our VariableStmt node is not 1:1 with typescript's, should we change it? Would that remove this special case code?

Reply:
We can, it would be to just change "name" from a string to an `Identifier`.